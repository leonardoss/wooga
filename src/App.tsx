import React, { FC } from 'react';

import Form from './components/Form/form.component';

import './App.scss';

const options = {
  action: '/foobar',
  method: 'post',
  layout: 'vertical',
  fields: [
    {
      type: 'text',
      label: 'Name',
      name: 'someName',
      validation: [
        {
          required: {
            value: true,
            msg: 'field required',
          },
        },
        {
          fieldStartsWith: {
            value: 'wooga.name',
            msg: 'expected field starting with: wooga.name',
          },
        },
      ],
    },
    {
      type: 'text',
      label: 'Name 2',
      name: 'someName 2',
      validation: [
        {
          required: {
            value: true,
            msg: 'field required',
          },
        },
      ],
    },
  ],
  submitButton: {
    label: 'Send data',
  },
  onSubmitForm: (values: any) => {
    console.log('--+ FORM SUBMITED: values on form: ', values);
  },
};

const App: FC = () => {
  return (
    <div className="App">
      <h1>Wooga - Frontend engineer technical test</h1>
      <Form {...options} />
    </div>
  );
};

export default App;
