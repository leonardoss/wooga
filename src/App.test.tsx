import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import App from './App';

describe('Testing App', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
  });
  it('App renders correctly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders title', () => {
    render(<App />);
    expect(screen.getByText('Wooga - Frontend engineer technical test')).toBeInTheDocument();
  });
});
