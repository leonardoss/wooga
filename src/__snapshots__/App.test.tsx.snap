// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App App renders correctly 1`] = `
<div
  className="App"
>
  <h1>
    Wooga - Frontend engineer technical test
  </h1>
  <form
    action="/foobar"
    className="layout-vertical"
    method="post"
    onSubmit={[Function]}
  >
    <div
      className="form-control"
    >
      <label
        className="form-label"
        htmlFor="someName"
      >
        Name
      </label>
      <input
        className="input-text"
        name="someName"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-control"
    >
      <label
        className="form-label"
        htmlFor="lastName"
      >
        Last Name
      </label>
      <input
        className="input-text"
        name="lastName"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-control"
    >
      <input
        className="btn btn-submit"
        type="submit"
        value="Send data"
      />
    </div>
  </form>
</div>
`;
